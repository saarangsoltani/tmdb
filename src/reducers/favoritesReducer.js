import {
  ADD_FAVORITE_ITEM,
  REMOVE_FAVORITE_ITEM
} from "../actions/types";
import { loadState } from "../localStorage";

let initialState = loadState("favorites") || [];

export const favoritesReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_FAVORITE_ITEM:
      return insertArrayItem(state, action.payload);
    case REMOVE_FAVORITE_ITEM:
      return removeArrayItem(state, action.payload);
    default:
      return state;
  }
};

function insertArrayItem(array, item) {
  let newArray = array.slice();
  newArray.push(item);
  return newArray;
}

function removeArrayItem(array, itemId) {
  let newArray = array.slice();
  var index = array.findIndex(fav => fav.id === itemId);
  newArray.splice(index, 1);
  return newArray;
}
