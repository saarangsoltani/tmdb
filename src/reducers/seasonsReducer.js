import { SET_SELECTED_SEASON } from "../actions/types";
let initialState = { selectedSeason: {episodes: []} };

export const seasonsReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_SELECTED_SEASON:
      return Object.assign({}, state, { selectedSeason: action.payload });
    default:
      return state;
  }
};
