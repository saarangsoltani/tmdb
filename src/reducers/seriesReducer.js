import {
  SET_DISCOVERED_SERIES,
} from "../actions/types";

let initialState = {
  selectedSeries: { seasons: [] },
  discovered: { page: 1, total_results: 0, total_pages: 0, results: [] }
};

export const seriesReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_DISCOVERED_SERIES:
      return Object.assign({}, state, { discovered: action.payload });
    default:
      return state;
  }
};
