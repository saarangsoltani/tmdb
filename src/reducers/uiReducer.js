import { API_START, API_END, SORT_SERIES } from "../actions/types";

let initialState = {
  selectedSortOrder: "popularity.desc",
  isLoading: false
};

export const uiReducer = (state = initialState, action) => {
  switch (action.type) {
    case SORT_SERIES:
      return Object.assign({}, state, { selectedSortOrder: action.payload });
    case API_START:
      return Object.assign({}, state, { isLoading: true });
    case API_END:
      return Object.assign({}, state, { isLoading: false });
    default:
      return state;
  }
};
