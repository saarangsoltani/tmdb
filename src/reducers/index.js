import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";
import { seriesReducer } from "./seriesReducer";
import { genresReducer } from "./genresReducer";
import { uiReducer } from "./uiReducer";

export default history =>
  combineReducers({
    router: connectRouter(history),
    series: seriesReducer,
    genres: genresReducer,
    ui: uiReducer
  });
