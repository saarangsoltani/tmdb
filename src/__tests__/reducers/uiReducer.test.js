import { uiReducer } from "../../reducers/uiReducer";
import {
  API_START,
  API_END,
  SORT_SERIES,
  FILTER_BY_GENRES
} from "../../actions/types";

let initialState = uiReducer(undefined, { type: "INIT" });

describe("reducers/ui", () => {
  it("should toggle loading state on API_START and API_END", () => {
    expect(uiReducer(initialState, { type: API_START })).toMatchSnapshot();
    expect(uiReducer(initialState, { type: API_END })).toMatchSnapshot();
  });

  it("should change the sort key on SORT_SERIES", () => {
    expect(
      uiReducer(initialState, { type: SORT_SERIES, payload: "revenue.asc" })
    ).toMatchSnapshot();
  });

  it("should change genres key on SORT_SERIES", () => {
    expect(
      uiReducer(initialState, {
        type: FILTER_BY_GENRES,
        payload: ["animation"]
      })
    ).toMatchSnapshot();
  });
});
