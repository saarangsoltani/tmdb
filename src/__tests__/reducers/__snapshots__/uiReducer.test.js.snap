// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/series should change the sort key on SORT_SERIES 1`] = `
Object {
  "isLoading": false,
  "selectedSortOrder": "revenue.asc",
}
`;

exports[`reducers/series should toggle loading state on API_START and API_END 1`] = `
Object {
  "isLoading": true,
  "selectedSortOrder": "popularity.desc",
}
`;
